bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/GenericEventTriggerHandler.java

```diff
@@ -87,6 +91,15 @@ public void receive(Event event) {
             if (!event.getTopic().contains(source)) {
                 return;
             }
+            // if the optional parameter "eventOnChannel" is set different from the defaultValue (""), compare it to the
+            // received event on the channel
+            if (eventOnChannel != null && !eventOnChannel.equals("")) {
+                if (event instanceof ChannelTriggeredEvent) {
```

Comment:
I think it is not good idea to add dependencies to the  third party libs for such check.